ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -305,9 +321,58 @@ void close(Exception exception) {
   private void closeRequest() {
     if (request != null && request.isOpen()) {
       request.close();
+      evaluatePerformanceAndUpdateMetrics();
     }
   }
 
+  private void evaluatePerformanceAndUpdateMetrics() {
+    RestRequestMetricsTracker restRequestMetricsTracker = request.getMetricsTracker();
+    long roundTripTimeInMs = restRequestMetricsTracker.getRoundTripTimeInMs();
+    long timeToFirstBytesInMs = restRequestMetricsTracker.getTimeToFirstByteInMs();
+    long totalOutboundBytes = totalBytesReceived.get();
+    long totalInboundBytes = request.getBytesReceived();
+    RestMethod method = request.getRestMethod();
+    Map<PerformanceIndex, Long> requestPerfToCheck = new HashMap<>();
+    responseStatus = errorResponseStatus == null ? responseStatus : errorResponseStatus;
```

Comment:
this statement is indispensable because in some error cases, the request is closed before `responseStatus` gets populated.